voicewatch/voicewatch.py

```diff
@@ -0,0 +1,98 @@
+        self.alertCache = {}
+        default_guild_config = {
+            "logchannel": "",  # Channel to send alerts to
+            "min_joined_hours": 8760  # Default to one year so the mods set it up :)
+        }
+
+        self.config.register_guild(**default_guild_config)
+
+    @checks.admin()
+    @commands.group("voicewatch", aliases=["vw"], pass_context=True)
+    async def _voicewatch(self, ctx: commands.Context):
+        pass
+
+    @_voicewatch.command(name="time")
+    async def _time(self, ctx: commands.Context, hours: str):
+        """Set/update the minimum hours users must be in the server before without triggering an alert."""
+        try:
+            hrs = int(hours)
+            await self.config.guild(ctx.guild).min_joined_hours.set(hrs)
+            await ctx.send("✅ Time requirement successfully updated!")
+        except ValueError:
+            await ctx.send("Error: Non-number hour argument supplied.")
+
+    @_voicewatch.command(name="logchannel")
+    async def _logchannel(self, ctx: commands.Context, channel: Optional[discord.TextChannel]):
+        """Set/update the channel to send voice activity alerts to."""
+
+        chanId = ctx.channel.id
+        if channel:
+            chanId = channel.id
+        await self.config.guild(ctx.guild).logchannel.set(chanId)
+        await ctx.send("✅ Alert channel successfully updated!")
+
+    @commands.Cog.listener()
+    async def on_voice_state_update(self, member: discord.Member,
+                                     before: discord.VoiceState, after: discord.VoiceState):
+        if member.id in self.alertCache and self.alertCache[member.id]:  # Ignore if we've already alerted on this user. Can probably find a better way.
+            # Todo: Make guild-aware
+            return
+        else:
+            self.alertCache[member.id] = False
+
+        if after is None or after.channel is None:  # Check if we're missing the after data or associated channel.
+            return
+
+        allowable = member.joined_at + timedelta(hours=await self.config.guild(after.channel.guild).min_joined_hours())
+        if datetime.now(timezone.utc) < allowable:
+            if after.self_stream or after.self_video:
+                self.alertCache[member.id] = True  # Update the cache to indicate we've alerted on this user.
+                # Credit: Taken from report Cog
+                log_id = await self.config.guild(after.channel.guild).logchannel()
+                log = None
+                if log_id:
+                    log = member.guild.get_channel(log_id)
+                if not log:
+                    # Failed to get the channel
+                    return
+
+                data = self.make_alert_embed(member, after.channel)
+
+                # Alert level logic added
+                mod_pings = " ".join(
+                    [i.mention for i in log.members if not i.bot and str(i.status) in ["online", "idle"]])
```

Comment:
```suggestion
                    i.mention for i in log.members if not i.bot and str(i.status) in ["online", "idle"])
```
We can make this a generator so that it doesn't get iterated twice 🎉